src/types/coefficient.rs

```diff
@@ -16,6 +16,67 @@ pub enum Sign {
     Positive,
 }
 
+/// A simple wrapper type around a [`UInt`]. Users are not expected to construct
+/// instances of `Magnitude` directly; it is provided for conversion ergonomics.
+///
+/// Signed integer types cannot implement `Into<UInt>`. Instead, they implement
+/// `TryInto<UInt>` and report an error if the input integer is negative.
+///
+/// Signed integers can infallibly implement `Into<Magnitude>` by using their
+/// absolute value.
+pub struct Magnitude(UInt);
```

Comment:
🗺️ Previously, as described in #569, signed integers being converted to `UInt` would use their absolute value. This wrapper type offers the same behavior, but is narrowly used in APIs where that wouldn't be surprising.